docs/xshareclasses.md

```diff
@@ -239,6 +239,18 @@ case, the VM continues without using shared classes.
 
 : Destroys all caches that are unused for the time that is specified before loading shared classes. This option is not a utility option because it does not cause the VM to exit. On Windows systems, which have NTFS file systems, the `expire` option is accurate to the nearest hour.
 
+### `extraStartupHints`
+
+        -Xshareclasses:extraStartupHints=<number>
+
+: where, `<number>` is greater than or equal to 0.
+
+: Adjusts the number of startup hints that can be stored in a shared cache. By default, you can store only up to 64 startup hints in a shared cache. This count decrements by 1 each time a startup hint is stored. After the count reaches 0, no more hints can be added to the cache. You can use `-Xshareclasses:extraStartupHints=<number>` to adjust this count as needed. For example, `-Xshareclasses:extraStartupHints=0` prevents any new hints to be stored and `-Xshareclasses:extraStartupHints=10` allows 10 more new hints in addition to the default number of 64 startup hints.
+
+: You can use [`-Xshareclasses:printStats=startuphint`](#printstats-cache-utility) to check how many startup hints are already stored in the shared cache or [`-Xshareclasses:printDetails`](#printdetails-cache-utility) to check how many more hints can be stored in the cache.
+
+: The information related to the number of additional startup hints that can be stored is found on the line `# Additional startup hints allowed`.
```

Comment:
Since this is related to printdetails, seems it should be part of the previous paragraph and the previous sentence restructured to make it clear it's referring to printdetails.